### PR TITLE
mariadDB4j-maven-plugin goal start and stop to run integration tests …

### PR DESCRIPTION
**Pull request by duttonw**
This is used to bring up a database up before a web application is booted.
Where said application does not have easy hooks to start an embedded database easily
and is only used to emulate higher environment without need to deal with differinces
in database language or configuration options.

Can be used in combination with build-helper:reserve-network-port and spring-boot:start,stop
ensure that reserve-network-port is before this plugin and spring-boot is after, else
the spring application will be started before the database (maven xml ordering matters)

This could also be forked with jmx server management like our spring-boot start/stop does it controls.

Please let me know what you think, still needs a bit more work. 

I need to verify test this on another system or have time to downgrade my laptop. (mac brew flask install java upgraded to java 10 on and its not cooperating :'( )

### CODE REVIEW

**Comment by vorburger** [comment]
@duttonw how does this relate to https://github.com/mike10004/mariadb4j-maven-plugin ? 

Completely re-implementation by coincidence? Or copy/paste fork of that project by @mike10004 ?

**Comment by duttonw** [comment]
Never saw that project... :$. Similar but different style. It’s not been looked after so could be a good thing to incorperate into the main repo as you already have spring helpers etc. also seeing how spring boot has everything in one repo it could be more  beneficial to know they everything works together.

**Comment by mike10004** [comment]
I did have to abandon that plugin project, unfortunately; good luck you to you in your efforts

File: mariaDB4j-maven-plugin/pom.xml [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+                                <id>generate-site</id>
+                                <goals>
+                                    <goal>jar</goal>
+                                </goals>
+                            </execution>
+                        </executions>
+                    </plugin>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-invoker-plugin</artifactId>
```
[the remark this answers is not in the document]
Comment (duttonw) [reply mariaDB4j-maven-plugin/pom.xml]: was basing my profile's on the spring-boot maven plugin profile's

File: mariaDB4j-maven-plugin/pom.xml [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+        <maven-invoker-plugin.version>3.1.0</maven-invoker-plugin.version>
+        <maven-jar-plugin.version>3.0.2</maven-jar-plugin.version>
+        <maven-javadoc-plugin.version>3.0.0</maven-javadoc-plugin.version>
+        <maven-resources-plugin.version>3.0.1</maven-resources-plugin.version>
+        <maven-shade-plugin.version>2.4.3</maven-shade-plugin.version>
+        <maven-site-plugin.version>3.6</maven-site-plugin.version>
+        <maven-source-plugin.version>3.0.1</maven-source-plugin.version>
+        <maven-surefire-plugin.version>2.22.0</maven-surefire-plugin.version>
+        <maven-war-plugin.version>3.1.0</maven-war-plugin.version>
+        <maven.version>3.1.1</maven.version>
```
[the remark this answers is not in the document]
Comment (duttonw) [reply mariaDB4j-maven-plugin/pom.xml]: spring-boot maven plugin allows 3.1.1 and upwards. So i'm basing it on that. Also many places may not be on the latest maven and you don't want to lock them out of this fantastic tool.

**Comment by duttonw** [comment]
I'll see if I have time to [subtree merge](https://git-scm.com/book/en/v1/Git-Tools-Subtree-Merging) https://github.com/mike10004/mariadb4j-maven-plugin into this project.

This may be done with me rebasing my work on top of his so that the history of what he started is not lost.

Also need to add more Mockito unit tests and plugin in a [code coverage checker](https://www.eclemma.org/jacoco/trunk/doc/maven.html) to ensure our level of testing starts up and [owasp dependancy checker](https://jeremylong.github.io/DependencyCheck/dependency-check-maven/index.html) to ensure the libraries in use are not flagged with vulnerabilities when the next build happens.

May also add Findbugs, pmd, and also git-commit-id-plugin so that we have a git.properties in all artifact types.

**Comment by vorburger** [comment]
> Also need to add more Mockito unit tests 

just FTR I'm happy to merge this PR as it is without additional tests, if the review feedback is OK; I'm going to re-review hopefully merge it when I get a minute this week.

> and plugin in a code coverage checker to ensure our level of testing starts up and owasp dependancy checker to ensure the libraries in use are not flagged with vulnerabilities when the next build happens.
> May also add Findbugs, pmd, and also git-commit-id-plugin so that we have a git.properties in all artifact types.

that would all be cool but better in separate new PRs, not mixed into this PR here?

File: mariaDB4j-maven-plugin/src/main/resources/META-INF/m2e/lifecycle-mapping-metadata.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<lifecycleMappingMetadata>
+	<pluginExecutions>
+		<pluginExecution>
+		</pluginExecution>
```
[the remark this answers is not in the document]
Comment (vorburger) [reply mariaDB4j-maven-plugin/src/main/resources/META-INF/m2e/lifecycle-mapping-metadata.xml]: so what about this? I'll merge it if no other objections; great if it does work, if NOK, can be fixed in future PRs.

**Comment by vorburger** [comment]
@duttonw thanks again for contributing. As these things go, I'll be tagging you for help with any issues any end-users hit with this part of the code, OK for you?

**Pull request merged**